include/small/util.h

```diff
@@ -63,6 +68,25 @@
 #  define SMALL_ALWAYS_INLINE inline
 #endif
 
+#if !defined(__cplusplus) && !defined(static_assert)
+#  define static_assert _Static_assert
+#endif
+
+#ifndef lengthof
+#  define lengthof(array) (sizeof (array) / sizeof ((array)[0]))
+#endif
+
+#define small_xmalloc(size)							\
+	({									\
+		void *ret = malloc(size);					\
+		if (small_unlikely(ret == NULL)) {				\
+			fprintf(stderr, "Can't allocate %zu bytes at %s:%d",	\
+				(size_t)(size), __FILE__, __LINE__);		\
+			exit(EXIT_FAILURE);					\
+		}								\
```

Comment:
Nit: You could avoid the dependency on `stdio.h` by calling `small_xmalloc_fail` defined in `uitl.c` on allocation failure.

Reply:
Done.